core/src/main/scala/kafka/server/metadata/BrokerMetadataPublisher.scala

```diff
@@ -258,6 +270,15 @@ class BrokerMetadataPublisher(conf: KafkaConfig,
     // recovery-from-unclean-shutdown if required.
     logManager.startup(metadataCache.getAllTopics())
 
+    // Make the LogCleaner available for reconfiguration. We can't do this prior to this
+    // point because LogManager#startup creates the LogCleaner object, if
+    // log.cleaner.enable is true.
+    //
+    // TODO: it would probably be better to make log.cleaner.enable dynamically
+    // configurable as well. It would also be cleaner to unconditionally create the log
+    // cleaner object in the LogManager constructor even if we never start it.
```

Comment:
Should we just create an issue that documents this implementation suggestions and remove this comment?

Reply:
OK. I created https://issues.apache.org/jira/browse/KAFKA-13610 for this. I will reference that JIRA here.